src/frontend/src/flows/recovery/setupRecovery.ts

```diff
@@ -99,8 +99,24 @@ export const setupPhrase = async (
   );
 
   const phrase = userNumber.toString(10) + " " + seedPhrase;
-  await displaySeedPhrase(phrase);
-  await confirmSeedPhrase({ phrase });
+
+  // Loop until the user has confirmed the phrase
+  for (;;) {
+    await displaySeedPhrase(phrase);
+
+    const result = await confirmSeedPhrase({ phrase });
+    // User has confirmed, so break out of the loop
+    if (result === "confirmed") {
+      break;
+    }
+
+    // User has clicked the back button, so we retry
+    if (result === "back") {
+      continue;
```

Comment:
Again, I'm not 100% fan of the infinite loop + `continue` combo, but I haven't found a better (non-recursive) way. Will try to fix this up when adding wizard components.